test/component_test.rb

```diff
@@ -102,18 +104,82 @@ class ComponentTest < ActiveSupport::TestCase
     component_class = Class.new(Components::Component) do
       element :foo
     end
-    component = component_class.new(:view)
+    component = component_class.new(view_class.new)
     assert_nil component.foo
   end
 
   test "get element with multiple true when not set" do
     component_class = Class.new(Components::Component) do
       element :foo, multiple: true
     end
-    component = component_class.new(:view)
+    component = component_class.new(view_class.new)
     assert_equal [], component.foos
   end
 
+  test "initialize with given attribute and successfull validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+    end
+
+    assert_nothing_raised { component_class.new(view_class.new, foo: "bar") }
+  end
+
+  test "initialize without attribute and failing validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+    end
+
+    e = assert_raises(ActiveModel::ValidationError) { component_class.new(view_class.new) }
+    assert_equal "Validation failed: Foo can't be blank", e.message
+  end
+
+  test "initialize with default value and successfull validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo, default: "bar"
+      validates :foo, presence: true
+    end
+
+    assert_nothing_raised { component_class.new(view_class.new) }
+  end
+
+  test "initialize element and successfull validation" do
+    component_class = Class.new(Components::Component) do
+      element :foo do
+        attribute :label
+        attribute :bar
+
+        validates :bar, presence: true
+      end
+    end
+
+    assert_nothing_raised do
+      component_class.new(view_class.new, {}) do |c|
+        c.foo(bar: "lalal") { "something" }
+      end
+    end
+  end
+
+  test "initialize element and failing validation" do
+    component_class = Class.new(Components::Component) do
+      element :foo do
+        attribute :label
+        attribute :bar
+
+        validates :bar, presence: true
+      end
+    end
+
+    e = assert_raises(ActiveModel::ValidationError) do
+      component_class.new(view_class.new, {}) do |c|
+        c.foo(label: "lalal") { "label" }
```

Comment:
Perhaps no need to have a `label` attribute here either.

Reply:
Removed label attribute here too-